platforms/ide/ide-plugins/src/main/java/org/gradle/plugins/ide/idea/internal/IdeaScalaConfigurer.java

```diff
@@ -138,22 +138,28 @@ private static Iterable<File> getIdeaModuleLibraryDependenciesAsFiles(IdeaModule
         return files;
     }
 
-    @SuppressWarnings({"deprecation", "unused"})
-    private static ProjectLibrary createScalaSdkLibrary(Project scalaProject, Iterable<File> files, boolean useScalaSdk, IdeaModule ideaModule) {
-        ScalaRuntime runtime = scalaProject.getExtensions().findByType(ScalaRuntime.class);
-        if (runtime != null) {
-            FileCollection scalaClasspath = runtime.inferScalaClasspath(files);
-            File compilerJar = runtime.findScalaJar(scalaClasspath, "compiler");
-            if (compilerJar == null) {
-                compilerJar = runtime.findScalaJar(scalaClasspath, "compiler_3");
-            }
-            String scalaVersion = compilerJar != null ? runtime.getScalaVersion(compilerJar) : DEFAULT_SCALA_PLATFORM_VERSION;
-            return createScalaSdkFromScalaVersion(scalaVersion, scalaClasspath, useScalaSdk);
-        } else {
-            // One of the Scala plugins is applied, but ScalaRuntime extension is missing or the ScalaPlatform is undefined.
-            // we can't create a Scala SDK without either one
-            return null;
+    private static ProjectLibrary createScalaSdkLibrary(Project scalaProject, boolean useScalaSdk) {
+        ScalaPluginExtension scalaPluginExtension = scalaProject.getExtensions().getByType(ScalaPluginExtension.class);
```

Comment:
Is this guaranteed to always be there?

Reply:
The `ScalaBasePlugin` creates this extension unconditionally. This code is only executed on projects returned by `findProjectsApplyingIdeaAndScalaPlugins`, and therefore projects which apply the `ScalaBasePlugin`